neonvm/runner/container-mgr/main.go

```diff
@@ -139,14 +143,201 @@ func main() {
+		LOOPBACK_ADDRESS int32 = 127 + (0 << 8) + (0 << 16) + (1 << 24) // bswap(127.0.0.1)
+	)
+
+	insns := asm.Instructions{
+		// R1 contains a pointer to an __sk_buff (defined in bpf.h)
+		// Load length, and remote_ip4 into R6 and R7
+		asm.LoadMem(asm.R6, asm.R1, LEN_OFFSET, asm.Word),
+		asm.LoadMem(asm.R7, asm.R1, REMOTE_IP4_OFFSET, asm.Word),
+	}
+
+	if ENABLE_DEBUG_PRINTING {
+		insns = append(insns,
+			// Store the string "%d\n\0" to the stack
+			asm.Mov.Reg(asm.R1, asm.RFP),
+			asm.Add.Imm(asm.R1, -4),
+			asm.StoreImm(asm.R1, 0, 0x000a6425, asm.Word),
+			// Store the size of the string in R2 (which is 4
+			// including the null terminator)
+			asm.Mov.Imm(asm.R2, 4),
+			// Store the value we want to print (the IP address) in R3
+			asm.Mov.Reg(asm.R3, asm.R7),
+			asm.FnTracePrintk.Call(),
+		)
+	}
+
+	insns = append(insns,
+		// Check if remote address is one of:
+		// - empty address (0.0.0.0)
+		// - loopback address (127.0.0.1)
+		// - class A private network (10.0.0.0/8)
+		// - class B private network (172.16.0.0/12)
+		// - class C private network (192.168.0.0/16)
+		// If it is any of these, exit
+		asm.JEq.Imm32(asm.R7, 0, "exit"),
+		asm.JEq.Imm32(asm.R7, LOOPBACK_ADDRESS, "exit"),
+		asm.Mov.Reg32(asm.R8, asm.R7),
+		asm.And.Imm32(asm.R8, CLASS_A_MASK),
+		asm.JEq.Imm32(asm.R8, CLASS_A_ADDRESS, "exit"),
+		asm.Mov.Reg32(asm.R8, asm.R7),
+		asm.And.Imm32(asm.R8, CLASS_B_MASK),
+		asm.JEq.Imm32(asm.R8, CLASS_B_ADDRESS, "exit"),
+		asm.Mov.Reg32(asm.R8, asm.R7),
+		asm.And.Imm32(asm.R8, CLASS_C_MASK),
+		asm.JEq.Imm32(asm.R8, CLASS_C_ADDRESS, "exit"),
+
+		// Load the map into R1
+		// By the way: isn't it funny how we're putting a file descriptor
+		// here? Even though this program is getting shared among multiple
+		// processes?
+		// Well, it turns out that when this program gets loaded into the kernel
+		// there's a special pass that converts these LoadMapPtr calls from FDs
+		// into pointers to kernel memory. Pretty nifty!
+		asm.LoadMapPtr(asm.R1, counters.FD()),
+
+		// bpf_map_lookup_elem takes a pointer to the key, so we
+		// have to spill the key to the stack. The key is the
+		// network direction (i.e. 0 for ingress and 1 for egress).
+		asm.Mov.Reg(asm.R2, asm.RFP),
+		asm.Add.Imm(asm.R2, -4),
+		asm.StoreImm(asm.R2, 0, int64(direction), asm.Word),
+		// Now we have a pointer to the map in R1 and
+		// a pointer to the key in R2.
+		asm.FnMapLookupElem.Call(),
+		asm.JEq.Imm(asm.R0, 0, "exit"),
+		// R0 now holds a pointer to the counter
+		// Atomically add the packet length (stored in R6) to it
+		asm.StoreXAdd(asm.R0, asm.R6, asm.DWord),
+		asm.Mov.Imm(asm.R0, 1).WithSymbol("exit"),
+		asm.Return(),
+	)
+
+	//nolint:exhaustruct // This has a whole bunch of optional configuration options that we don't care about
+	prog, err := ebpf.NewProgram(&ebpf.ProgramSpec{
+		Name:         "neon_network_monitor",
+		Type:         ebpf.CGroupSKB,
+		License:      "GPL",
```

Comment:
Ah yeah, I changed it to GPL  for the printing thing because I thik I originally had "Apache 2.0" and it was rejecting it. Changed it to "Apache-2.0" and I think it works now?